frontend/app/controllers/Healthcheck.scala

```diff
@@ -1,34 +1,70 @@
 package controllers
 
-import com.github.nscala_time.time.Imports._
+import akka.agent.Agent
+import com.gu.membership.model.{FriendTierPlan, PaidTierPlan, StaffPlan}
+import com.gu.membership.salesforce.Tier.{Partner, Patron, Supporter}
 import com.gu.monitoring.CloudWatchHealth
 import play.api.Logger
+import play.api.libs.concurrent.Akka
+import play.api.libs.concurrent.Execution.Implicits._
 import play.api.mvc.{Action, Controller}
 import services.{GuardianLiveEventService, TouchpointBackend}
+import play.api.Play.current
 
-case class Test(name: String, ok: () => Boolean)
+import scala.concurrent.Future
+
+case class TestCase(name: String, exec: () => Future[Boolean]) {
+  def failedTest() = Logger.warn(s"Test $name failed, health check will fail")
+
+  def ok() = {
+    val result = exec()
+    result.map(passed => if (!passed) failedTest())
```

Comment:
they are logging a failed test which can fail for one of two reasons: if the result of the test is false or the test has thrown an exception